editor/src/components/canvas/canvas-strategies/strategies/grid-helpers.ts

```diff
@@ -400,3 +403,51 @@ function asMaybeNamedAreaOrValue(
   }
   return value
 }
+
+export function getGridCellBoundsFromCanvas(
+  cell: ElementInstanceMetadata,
+  canvasScale: number,
+  canvasOffset: CanvasVector,
+) {
+  const cellFrame = cell.globalFrame
+  if (cellFrame == null || isInfinityRectangle(cellFrame)) {
+    return null
+  }
+
+  const canvasFrameWidth = cellFrame.width * canvasScale
+  const canvasFrameHeight = cellFrame.height * canvasScale
+
+  const cellOriginPoint = offsetPoint(
+    canvasPointToWindowPoint(cellFrame, canvasScale, canvasOffset),
+    windowPoint({ x: 5, y: 5 }),
```

Comment:
why 5, btw?

Reply:
ah, because it's the minimum for which it'll always work fine :) it's 2 + 2 per side, plus 1 for good measure

Reply:
[`a67d57d` (#6298)](https://github.com/concrete-utopia/utopia/pull/6298/commits/a67d57d157f0787edd1afdc2e133d7aed914ce80) refactored to a const